mava/wrappers/jumanji.py

```diff
@@ -12,35 +12,87 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Tuple
+from typing import Callable, Dict, Tuple
 
 import chex
 import jax.numpy as jnp
 from jumanji import specs
-from jumanji.environments.routing.robot_warehouse import RobotWarehouse
+from jumanji.env import Environment
 from jumanji.types import TimeStep
 from jumanji.wrappers import Wrapper
 
 from mava.types import Observation, State
 
 
-class RwareWrapper(Wrapper):
-    """Multi-agent wrapper for the Robotic Warehouse environment."""
+class MultiAgentWrapper(Wrapper):
+    """
+    Multi-agent wrapper for environments with configurable reward handling.
 
-    def __init__(self, env: RobotWarehouse):
+    Args:
+        env (Environment): The base environment.
+        use_individual_reward (bool): If True, the network uses the list of different rewards given.
+        aggregate_rewards (bool): If True, aggregates rewards across agents.
+        reward_aggregation_function (str): The function for aggregating rewards ("sum" or "mean").
+    """
+
+    def __init__(
+        self,
+        env: Environment,
+        use_individual_reward: bool = False,
+        aggregate_rewards: bool = False,
+        reward_aggregation_function: str = "sum",
```

Comment:
I think we shouldn't have these options, we should decide if we want individual/aggregated rewards and use the option we want in each separate wrapper

Reply:
If we are doing a different wrapper per env then hard setting the aggregation is fine.

Reply:
For the aggregation  I let the user choose whether to use the list of individual rewards (default setting) or  set the `aggregate_rewards=True` in lbf config file.

Reply:
This is good, but I don't think we should give an option for _what_ aggregation to use. We should just hardcode it in the wrapper to be the same as what other benchmarks do for each environment.